resources/plugins/simln/plugin.py

```diff
@@ -0,0 +1,224 @@
+
+PLUGIN_DIR_TAG = "plugin_dir"
+
+
+class PluginError(Exception):
+    pass
+
+
+log = logging.getLogger(MISSION)
+log.setLevel(logging.DEBUG)
+console_handler = logging.StreamHandler()
+console_handler.setLevel(logging.DEBUG)
+formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+console_handler.setFormatter(formatter)
+log.addHandler(console_handler)
+
+
+class PluginContent(Enum):
+    ACTIVITY = "activity"
+
+
+@click.group()
+@click.pass_context
+def simln(ctx):
+    """Commands for the SimLN plugin"""
+    ctx.ensure_object(dict)
+    plugin_dir = Path(__file__).resolve().parent
+    ctx.obj[PLUGIN_DIR_TAG] = Path(plugin_dir)
+
+
+@simln.command()
+@click.argument("plugin_content", type=str)
+@click.argument("warnet_content", type=str)
+@click.pass_context
+def entrypoint(ctx, plugin_content: str, warnet_content: str):
+    """Plugin entrypoint"""
+    plugin_content: dict = json.loads(plugin_content)
+    warnet_content: dict = json.loads(warnet_content)
+
+    hook_value = warnet_content.get(WarnetContent.HOOK_VALUE.value)
+
+    assert hook_value in {
+        item.value for item in HookValue
+    }, f"{hook_value} is not a valid HookValue"
+
+    if warnet_content.get(PLUGIN_ANNEX):
+        for annex_member in [annex_item for annex_item in warnet_content.get(PLUGIN_ANNEX)]:
+            assert annex_member in {
+                item.value for item in AnnexMember
+            }, f"{annex_member} is not a valid AnnexMember"
+
+    warnet_content[WarnetContent.HOOK_VALUE.value] = HookValue(hook_value)
+
+    _entrypoint(ctx, plugin_content, warnet_content)
+
+
+def _entrypoint(ctx, plugin_content: dict, warnet_content: dict):
+    """Called by entrypoint"""
+    # write your plugin startup commands here
+    activity = plugin_content.get(PluginContent.ACTIVITY.value)
+    activity = json.loads(activity)
+    print(activity)
+    _launch_activity(activity, ctx.obj.get(PLUGIN_DIR_TAG))
+
+
+@simln.command()
+def list_pod_names():
+    """Get a list of SimLN pod names"""
+    print([pod.metadata.name for pod in get_mission(MISSION)])
+
+
+@simln.command()
+@click.argument("pod_name", type=str)
+def download_results(pod_name: str):
+    """Download SimLN results to the current directory"""
+    dest = download(pod_name, source_path=Path("/working/results"))
+    print(f"Downloaded results to: {dest}")
+
+
+def _get_example_activity() -> list[dict]:
+    pods = get_mission(LIGHTNING_MISSION)
+    try:
+        pod_a = pods[1].metadata.name
+        pod_b = pods[2].metadata.name
+    except Exception as err:
+        raise PluginError(
+            "Could not access the lightning nodes needed for the example.\n Try deploying some."
+        ) from err
+    return [{"source": pod_a, "destination": pod_b, "interval_secs": 1, "amount_msat": 2000}]
+
+
+@simln.command()
+def get_example_activity():
+    """Get an activity representing node 2 sending msat to node 3"""
+    print(json.dumps(_get_example_activity()))
```

Comment:
this seems unecessary. simln can generate its own random activity. I also dont love how its used in the test: create a json file from the plugin then pass it back to the plugin? I feel like it should be more self contained, like `simln/plugin.py entrypoint --with-activity-pattern <some options>`

Reply:
I made this for users to quickly see how to create an activity and what an activity looks like. 

Users can run this to get an example activity based on their cluster: 

`./simln/plugin.py get-example-activity`

And it also allows this kind of thing:

`./plugins/simln/plugin.py launch-activity "$(./plugins/simln/plugin.py get-example-activity)"`

I can see how this might seem more limited than omitting the activity and using the sim-ln random paymnet feature.